pkg/reconciler/revision/resources/deploy_test.go

```diff
@@ -399,6 +454,10 @@ func TestMakePodSpec(t *testing.T) {
 	}{{
 		name: "user-defined user port, queue proxy have PORT env",
 		rev: revision("bar", "foo",
+			withPodSpec([]corev1.Container{{
```

Comment:
These fixtures here have become hard to understand with the additional of `withPodSpec` since further down there's another `revisionOption` that's modifying the container that you've built here